xbmc/windowing/windows/WinSystemWin32.cpp

```diff
@@ -464,19 +465,63 @@ bool CWinSystemWin32::ChangeResolution(RESOLUTION_INFO res)
     sDevMode.dmDisplayFlags = (res.dwFlags & D3DPRESENTFLAG_INTERLACED) ? DM_INTERLACED : 0;
     sDevMode.dmFields = DM_PELSWIDTH | DM_PELSHEIGHT | DM_DISPLAYFREQUENCY | DM_DISPLAYFLAGS;
 
-    // CDS_FULLSCREEN is for temporary fullscreen mode and prevents icons and windows from moving
-    // to fit within the new dimensions of the desktop
-    LONG rc = ChangeDisplaySettingsExW(details.DeviceNameW.c_str(), &sDevMode, NULL, CDS_FULLSCREEN, NULL);
-    if (rc != DISP_CHANGE_SUCCESSFUL)
+    // Windows 8 refresh rate workaround for 24.0, 48.0 and 60.0 Hz
+    if (CSysInfo::IsWindowsVersionAtLeast(CSysInfo::WindowsVersionWin8) && ((int)res.fRefreshRate == 24 || (int)res.fRefreshRate == 48 || (int)res.fRefreshRate == 60))
     {
-      CLog::Log(LOGERROR, "%s : ChangeDisplaySettingsEx failed with %d", __FUNCTION__, rc);
-      return false;
+      DEVMODEW sDevModeRegistry;
+      ZeroMemory(&sDevModeRegistry, sizeof(sDevModeRegistry));
+      sDevModeRegistry.dmSize = sizeof(sDevModeRegistry);
+      if (EnumDisplaySettingsW(details.DeviceNameW.c_str(), ENUM_REGISTRY_SETTINGS, &sDevModeRegistry))
+      {
+        CLog::Log(LOGDEBUG, "%s : Activating Windows 8+ workaround for refresh rate %d Hz", __FUNCTION__, (int)res.fRefreshRate);
+        bool resChanged = true;
+
+        // Set requested mode in registry without actually changing resolution
+        LONG rc = ChangeDisplaySettingsExW(details.DeviceNameW.c_str(), &sDevMode, NULL, CDS_UPDATEREGISTRY | CDS_NORESET, NULL);
+        if (rc != DISP_CHANGE_SUCCESSFUL)
+        {
+          CLog::Log(LOGERROR, "%s : ChangeDisplaySettingsEx (set registry) failed with %d", __FUNCTION__, rc);
+          return false;
+        }
+
+        // Change resolution based on registry setting
+        rc = ChangeDisplaySettingsExW(details.DeviceNameW.c_str(), NULL, NULL, CDS_FULLSCREEN, NULL);
+        if (rc != DISP_CHANGE_SUCCESSFUL)
+        {
+          CLog::Log(LOGERROR, "%s : ChangeDisplaySettingsEx (change resolution) failed with %d", __FUNCTION__, rc);
+          resChanged = false;
+        }
+
+        // Restore registry with original values
+        sDevModeRegistry.dmSize = sizeof(sDevModeRegistry);
+        sDevModeRegistry.dmDriverExtra = 0;
+        sDevModeRegistry.dmFields = DM_PELSWIDTH | DM_PELSHEIGHT | DM_DISPLAYFREQUENCY | DM_DISPLAYFLAGS;
+        rc = ChangeDisplaySettingsExW(details.DeviceNameW.c_str(), &sDevModeRegistry, NULL, CDS_UPDATEREGISTRY | CDS_NORESET, NULL);
+        if (rc != DISP_CHANGE_SUCCESSFUL)
+        {
+          CLog::Log(LOGERROR, "%s : ChangeDisplaySettingsEx (restore registry) failed with %d", __FUNCTION__, rc);
+        }
+
+        return resChanged;
+      }
     }
     else
```

Comment:
if EnumDisplaySettings is failed resolution will not be changed with no log message. I suggest to record to the log and try to change resolution with older variant.